src/en/infinityscans/src/eu/kanade/tachiyomi/extension/en/infinityscans/InfinityScans.kt

```diff
@@ -49,59 +47,106 @@ class InfinityScans : HttpSource() {
         add("Sec-Fetch-Dest", "empty")
         add("Sec-Fetch-Mode", "cors")
         add("Sec-Fetch-Site", "same-origin")
-        add("X-Requested-With", "XMLHttpRequest")
+        add("X-requested-with", "XMLHttpRequest")
     }.build()
 
     private val json: Json by injectLazy()
 
     // Popular
 
-    override fun popularMangaRequest(page: Int): Request =
-        searchMangaRequest(page, "", FilterList(SortFilter("popularity")))
+    override fun popularMangaRequest(page: Int): Request = fetchJson("api/ranking")
 
-    override fun popularMangaParse(response: Response): MangasPage =
-        searchMangaParse(response)
+    override fun popularMangaParse(response: Response): MangasPage {
+        val data = response.parseAs<ResponseDto<RankingResultDto>>().result
+        val entries = data.weekly
+            .map { it.toSManga(cdnHost) }
+
+        return MangasPage(entries, false)
+    }
 
     // Latest
 
-    override fun latestUpdatesRequest(page: Int): Request =
-        searchMangaRequest(page, "", FilterList(SortFilter("latest")))
+    override fun latestUpdatesRequest(page: Int): Request = fetchJson("api/comics")
 
-    override fun latestUpdatesParse(response: Response): MangasPage =
-        searchMangaParse(response)
+    override fun latestUpdatesParse(response: Response): MangasPage {
+        val data = response.parseAs<ResponseDto<SearchResultDto>>().result
+        runCatching { updateFilters(data) }
 
-    // Search
+        val entries = data.titles.sortedByDescending { it.updated }
+            .map { it.toSManga(cdnHost) }
 
-    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
-        val url = baseUrl.toHttpUrl().newBuilder()
-            .addPathSegments("ajax/comics")
-            .addQueryParameter("page", page.toString())
+        return MangasPage(entries, false)
+    }
 
-        filters.forEach { filter ->
-            when (filter) {
-                is SortFilter -> {
-                    url.addQueryParameter("sort", filter.selected)
-                }
-                is GenreFilter -> {
-                    filter.checked?.also {
-                        url.addQueryParameter("genre", it.joinToString("|"))
-                    }
-                }
-                is AuthorFilter -> {
-                    filter.checked?.also {
-                        url.addQueryParameter("author", it.joinToString("|"))
-                    }
+    // Search
+
+    override fun fetchSearchManga(
+        page: Int,
+        query: String,
+        filters: FilterList,
+    ): Observable<MangasPage> {
+        return client.newCall(fetchJson("api/comics"))
+            .asObservableSuccess()
+            .map { response ->
+                val data = response.parseAs<ResponseDto<SearchResultDto>>().result
```

Comment:
Consider extracting this scope into another function, so you don't have too deep indentation.

Reply:
Hm...
I don't see any meaning to extracting this part, as it doesn't duplicate between any function and it doesn't add any clarity to what the code does.

From my point of view this function is somewhat big, and maybe need to be broken up, but it is one whole entity.

Reply:
I just saw your other comment. In that case there's no straight forward way to do it.

> fetchSearchManga is impossible to revert as in parse method we don't know about query or filters.